torchvision/prototype/datasets/_builtin/sintel.py

```diff
@@ -0,0 +1,167 @@
+                split=("train", "test"),
+                pass_=("clean", "final"),
+            ),
+        )
+
+    def resources(self, config: DatasetConfig) -> List[OnlineResource]:
+        archive = HttpResource(
+            "http://sintel.cs.washington.edu/MPI-Sintel-complete.zip",
+            sha256="",
+        )
+        return [archive]
+
+    def _classify_train_test(self, data: Tuple[str, Any], *, config: DatasetConfig) -> bool:
+        path = pathlib.Path(data[0])
+        return config.split in str(path.parent)
+
+    def _classify_archive(self, data: Tuple[str, Any], *, config: DatasetConfig) -> Union[int, None]:
+        path = pathlib.Path(data[0])
+        if config.pass_ == path.parent.parent.name and path.suffix == ".png":
+            return 0
+        elif path.suffix == ".flo":
+            return 1
+        else:
+            return None
+
+    def _read_flo(self, file: IterDataPipe[Tuple[Any, io.IOBase]]) -> Any:
+        magic = file.read(4)
+        if magic != b"PIEH":
+            raise ValueError("Magic number incorrect. Invalid .flo file")
+        w = int.from_bytes(file.read(4), "little")
+        h = int.from_bytes(file.read(4), "little")
+        data = file.read(2 * w * h * 4)
+        data = np.frombuffer(data, dtype=np.float32)
+        return data.reshape(h, w, 2).transpose(2, 0, 1)
+
+    def _flows_key(self, data: Tuple[str, Any]) -> Tuple[str, int]:
+        path = pathlib.Path(data[0])
+        category = path.parent.name
+        idx = int(FILE_NAME_PATTERN.match(path.name).group("idx"))  # type: ignore[union-attr]
+        return category, idx
+
+    def _images_key(self, data: Tuple[Tuple[str, Any], Tuple[str, Any]]) -> Tuple[str, int]:
+        return self._flows_key(data[0])
+
+    def _collate_and_decode_sample(
+        self,
+        data: Tuple[Tuple[str, io.IOBase], Any],
+        *,
+        decoder: Optional[Callable[[io.IOBase], torch.Tensor]],
+        config: DatasetConfig,
+    ) -> Dict[str, Any]:
+        if config.split == "train":
+            flo, images = data
+            img1, img2 = images
+            flow_arr = self._read_flo(flo[1])
+            del images, flo
+        else:
+            # When split is `test`
+            img1, img2 = data
+
+        del data
+
+        path1, buffer1 = img1
+        path2, buffer2 = img2
+
+        return dict(
+            image1=decoder(buffer1) if decoder else buffer1,
+            image2=decoder(buffer2) if decoder else buffer2,
+            flow=flow_arr if config.split == "train" else None,
+        )
+
+    def _make_datapipe(
+        self,
+        resource_dps: List[IterDataPipe],
+        *,
+        config: DatasetConfig,
+        decoder: Optional[Callable[[io.IOBase], torch.Tensor]],
+    ) -> IterDataPipe[Tuple[Dict[str, Any], Dict[str, Any]]]:
+        dp = resource_dps[0]
+        archive_dp = ZipArchiveReader(dp)
+
+        curr_split = Filter(archive_dp, partial(self._classify_train_test, config=config))
```

Comment:
No need for `partial` here

```suggestion
        curr_split = Filter(archive_dp, self._classify_train_test, fn_kwargs=dict(split=split))
```